coop_memberspace/models/res_partner.py

```diff
@@ -0,0 +1,175 @@
+        for member in members_dont_have_user:
+            already_email = self.env["res.partner"].search(
+                [("email", "=", member.email), ("id", "!=", member.id)]
+            )
+            if already_email:
+                _logger.error(
+                    _(
+                        "Another user is already registered"
+                        + " using this email address."
+                    )
+                )
+                continue
+            sql = """
+                SELECT id
+                FROM res_users
+                WHERE login = '%s'
+                LIMIT 1
+            """ % (
+                member.email,
+            )
+            self.env.cr.execute(sql)
+            user = self.env.cr.fetchone()
+            if user:
+                user = Users.browse(user[0])
+            if not user:
+                vals.update(
+                    {
+                        "partner_id": member.id,
+                        "name": member.name,
+                        "login": member.email,
+                        "email": member.email,
+                        "image": member.image,
+                    }
+                )
+                # Users.with_context(no_reset_password=True).create(vals)
+                Users.create(vals)
+            elif user.active:
+                user.partner_id = member.id
+        return True
+
+    @api.model
+    def cron_create_user_for_members(self):
+        members = self.search(
+            [("is_member", "=", True), ("email", "!=", False)]
+        )
+        members.create_memberspace_user()
+
+    @api.model
+    def get_partner_sex_website(self):
+        if self.sex == "m":
+            return "Homme"
+        elif self.sex == "f":
+            return "Femme"
+        return "Autre"
+
+    # @api.model
+    # def fields_view_get(
+    #     self,
+    #     view_id=None,
+    #     view_type="form",
+    #     context=None,
+    #     toolbar=False,
+    #     submenu=False,
+    # ):
+    #     res = super(ResPartner, self).fields_view_get(
+    #         view_id=view_id,
+    #         view_type=view_type,
+    #         context=context,
+    #         toolbar=toolbar,
+    #         submenu=submenu,
+    #     )
+    #     if view_type == "form":
+    #         doc = etree.XML(res["arch"])
+    #         nodes = doc.xpath("//field[@name='email_pos_receipt']")
+    #         user = self.env.user
+    #         has_group_memberspace = user.has_group(
+    #             "coop_memberspace.group_memberspace"
+    #         )
+    #         if nodes:
+    #             if not has_group_memberspace or (
+    #                 has_group_memberspace and not user.active
+    #             ):
+    #                 nodes[0].set("invisible", "1")
+    #                 setup_modifiers(
+    #                     nodes[0], res["fields"]["email_pos_receipt"]
+    #                 )
+    #                 res["arch"] = etree.tostring(doc)
+    #     return res
```

Comment:
Hmmm did you remove this?
If there's no use for it, just remove the commented lines.

I wonder why they implemented this in fields_view_get instead of extending the views.